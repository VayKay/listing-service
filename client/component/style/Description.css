.title{
    color: #484848;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
    font-size: 32px;
    font-weight: 600;
    line-height: 1.123em;
    padding-top: 6px;
    padding-bottom: 6px
}
.location{
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
    font-size: 16px;
    font-weight: 400;
    line-height: 1.375em;
    color: #484848 ;
}
.summary{
    width: 100%;
    height: 240px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.tilo{
    display: flex;
    flex-direction: column;
    margin-left: 8px;
    margin-right: 8px;
}
.host {
    text-align: center;
    margin-left: 8px;
    margin-right: 8px;
}
.hostname{
    font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, "Helvetica Neue", sans-serif;
    font-size: 14px ;
    font-weight: 400 ;
    line-height: 1.28571em ;
    color: rgb(118, 118, 118) ;
    margin: 0px
}
.host img{
    border-radius: 50%;
    width: 64px;
    height: 64px;
}
.hls{
    font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, Helvetica Neue, sans-serif;
    font-size: 14px;
    line-height: 1.43;
    color: #484848;
    background-color: #fff;
    margin: 0;
    margin-top: 20px
}
.hl{
    display: flex;
    flex-direction: row;
}
.hlicon{
    width: 24px;
    vertical-align: top
}
.hlicon img{
    width: 20x;
    height: 20px;
}
.hltitle{
    margin: 0px ;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 16px ;
    font-weight: 600 ;
    line-height: 1.375em ;
    color: #484848 ;
    margin-bottom: 16px;
}
.hlcontent{
    margin: 0px ;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    font-size: 16px ;
    font-weight: 400 ;
    line-height: 1.375em ;
    color: #484848;
}
.linemg{
    margin-top: 24px;
    margin-bottom: 24px;
}
.line{
    width: 100%;
    border:0.5px #EBEBEB solid;
}
.descs{
    width: 100%;
    margin-top: 20px;
}
.generaldesc{
    margin-bottom: 14px;
}
.desctitle:last-child{
    margin-bottom: 0;
}
.readmore{
    margin-top: 8px;
    margin-bottom: 24px;
    color: #008489;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
    font-size: 16px;
    font-weight: 600;
    line-height: 1.375em;
    cursor: pointer;
    text-decoration: none;
    flex-direction: row;
    align-items: center;
}
.readmore:hover, #readless:hover, #fakedesclink:hover{
    text-decoration: underline;
}
.readless{
    display: flex;
    flex-direction: row;
    align-items: center;
}
.readmore svg, .readless svg{
    width: 10px;
    height: 10px;
    margin-left: 8px;
    display: block;
    fill: currentColor
}
.readless svg{
    transform: rotate(180deg)
}
.desctitle{
    margin: 0px ;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 16px ;
    font-weight: 600 ;
    line-height: 1.375em;
    color: #484848;
    margin-bottom:  8px;
}
.descs p{
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 16px ;
    font-weight: 400 ;
    line-height: 1.375em ;
    color: #484848;
}
.readless{
    margin-top: 8px;
    margin-bottom: 24px;
    color: #008489;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
    font-size: 16px;
    font-weight: 600;
    line-height: 1.375em;
    cursor: pointer;
    text-decoration: none;
}
.fakedesclink{
    margin-top: 0px;
    margin-bottom: 24px;
    color: #008489;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
    font-size: 16px;
    font-weight: 600;
    line-height: 1.375em;
    cursor: pointer;
    text-decoration: none;
}
.roomarrg{
    display: flex;
    flex-direction: column;
    width: 100%;
}
.roomarrg h2{
    margin: 0px ;
    margin-bottom: 16px;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 16px ;
    font-weight: 600 ;
    line-height: 1.375em ;
    color: #484848 ;
}
.arrg{
    overflow: hidden;
    position: relative;
    margin: 0 auto;
    width: 100%;
}
.arrgarea{
    display: flex;
    justify-content: flex-start;
    width: 200%;
    height: 147px;
    left: -100%;
    overflow: visible;
    transition: -ms-transform 0.5s ease 0s, -webkit-transform 0.5s ease 0s, transform 0.5s ease 0s !important;
    white-space: nowrap !important;
    
}
.brarrg{
    width:  171.33px !important;
    padding: 24px;
    margin-left: 8px;
    margin-right: 8px;
    border: 1px solid #EBEBEB;
    border-radius: 4px;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 16px ;
    font-weight: 600 ;
    line-height: 1.375em ;
    color: #484848;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.brarrg svg{
    width: 24px;
    height: 24px;
    margin-bottom: 24px;
}

.brtext{
    margin: 0;
}
.bedinroom{
    margin: 0px ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 16px ;
    font-weight: 200 ;
}
.arrgline{
    display: flex;
    flex-direction: row;
    align-items: center;
    position: relative;
    z-index: 0;
    margin-left: -8px;
    margin-right: -8px;
}
.btnNext{
    position: absolute;
    right: -8px;
    transform: translateY(-50%);
    z-index: 1;
    transform: rotate(270deg);
}
.btnPre{
    position: absolute;
    left: -8px;
    transform: translateY(-50%);
    z-index: 1;
    transform: rotate(90deg);
}

.btnNext span, .btnPre span{
    top:50%
}
.btnNext button, .btnPre button{ 
    width: 32px;
    height: 32px;
    display: inline-block !important;
    cursor: pointer;
    text-align: center;
    line-height: 1;
    border-radius: 50%;
    position: relative;
    touch-action: manipulation;
    box-shadow: rgba(0, 0, 0, 0.14) 0px 1px 1px 1px;
    border: 2px solid transparent;
    background-color: #fff
}
.btnNext svg, .btnPre svg{
    width: 10px;
    height: 10px;   
    display: inline-block;
    fill:currentColor
}


.poplistback{
    position: fixed ;
    z-index: 2000 ;
    top: 0px ;
    right: 0px ;
    bottom: 0px ;
    left: 0px ;
    overflow-y: auto ;
    background: rgba(255, 255, 255, 0.8);
    padding: 64px;
   
}
.poplist{
    vertical-align: middle;
    margin-left: auto;
    margin-right: auto;
    background-color: rgb(255, 255, 255) ;
    position: relative ;
    box-shadow: rgba(0, 0, 0, 0.2) 0px 1px 10px 0px;
    padding: 32px;
    width: 504px;
}

.poplisthead{
    margin-bottom: 24px;
}
.poplisthead button{
    padding: 20px;
    margin: -20px;
    cursor: pointer ;
    background-color: transparent ;
    display: block;
    border: none;
}
.poplisthead svg{
    width: 16px;
    height: 16px;
    display: block;
    fill: #767676;
}
.poplisttitle{
    margin: 0px;
    padding-top: 2px ;
    padding-bottom: 2px;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 24px ;
    font-weight: 600 ;
    line-height: 1.25em ;
    color: #484848 ;   
}


.categorytitle{
    overflow-wrap: break-word ;
    margin: 0px;
    margin-top: 24px;
    margin-bottom: 24px;
    font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, "Helvetica Neue", sans-serif !important;
    font-size: 18px ;
    font-weight: 600 ;
    line-height: 1.44444em ;
    color: rgb(72, 72, 72) ;
}
.categorycontent{
    margin: 0px ;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    font-size: 16px ;
    font-weight: 300 ;
    line-height: 1.375em ;
    color: #484848;
}
.categorycomment{
    margin-top: 8px;
    font-size: 14px;
    font-weight: 300 ;
    line-height: 1.2857142857142858em;   
}
.poplistitem div:last-child{
    margin-bottom: 8px;
}
.poplistitem div:last-child .linemg{
    display: none;
}

.poplistcontent div:last-child .categorycontent{
    text-decoration: line-through;
}

@media (max-width: 743px) {
    .poplistback{
      background: none;
      padding: 0px;
    }
    .poplist{
        width: 100%;
        margin: 0;
        background-color: rgb(255, 255, 255) ;
        position: absolute ;
    }
} 