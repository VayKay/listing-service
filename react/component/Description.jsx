import React from 'react';
const path = require('path');
class Description extends React.Component{
    constructor(props){
        super(props)
        this.state={
            // showreadmorediv: true,
            showmoredesc: false,   
            showscrolltonext: false,
            move: 0,   
        }
    }
    readmoredesc(){
        this.setState({
            showmoredesc: true
        })
    }
    readlessdesc(){
        this.setState({
            showmoredesc: false
        })
    }   
    getnext(){
        this.setState({
            move: this.state.move-(100/3)
        })
    }
    getpre(){
        this.setState({
            move: this.state.move+(100/3)
        })
    }
    render(){
        // console.log('location',this.props.location);
        var title = this.props.title;
        var location = this.props.location;
        var host = this.props.host;
        var highlights = this.props.highlights;
        var desc = this.props.desc;
        var detail = this.props.detail;
        if(detail){
            var br = detail.bedrmnum;
            var bed = detail.bednum;
            var guest = detail.guestmax;
        }
        // console.log('test',br)
        var bedoptions = ['1 queen bed','1 single bed','1 king bed']
        var arrgs= [];
        var bedicon;
        for(var i=1; i<=br; i++){
            var bedoption = bedoptions[Math.floor(Math.random()*bedoptions.length)]
            if(bedoption==='1 queen bed' || bedoption==='1 king bed'){
                bedicon = 'm23.96 14.81-2.96-7.41v-5.02a1.39 1.39 0 0 0 -1.39-1.38h-15.22c-.77 0-1.39.62-1.39 1.38v5.02l-2.96 7.41-.04.19v5.61c0 .64.43 1.17 1.01 1.33 0 .02-.01.04-.01.06v1.5a.5.5 0 0 0 1 0v-1.5h20v1.5a.5.5 0 0 0 1 0v-1.5c0-.02-.01-.04-.01-.06a1.39 1.39 0 0 0 1.01-1.33v-5.61zm-19.96-12.43c0-.21.17-.38.39-.38h15.22a.39.39 0 0 1 .39.39v4.61h-1v-1.61c0-.77-.62-1.39-1.39-1.39h-3.21c-.78 0-1.4.62-1.4 1.39v1.61h-2v-1.61c0-.77-.62-1.39-1.39-1.39h-3.22c-.77 0-1.39.62-1.39 1.39v1.61h-1zm14 3.01v3.21a.39.39 0 0 1 -.39.39h-3.21a.39.39 0 0 1 -.4-.38v-3.22a.39.39 0 0 1 .39-.39h3.21a.39.39 0 0 1 .39.39zm-8 0v3.21a.39.39 0 0 1 -.39.4h-3.22a.39.39 0 0 1 -.39-.39v-3.22a.39.39 0 0 1 .39-.39h3.21a.39.39 0 0 1 .39.39zm-6.16 2.61h1.16v.61c0 .77.62 1.39 1.39 1.39h3.21c.78 0 1.4-.62 1.4-1.39v-.61h2v .61c0 .78.62 1.39 1.39 1.39h3.21c.78 0 1.4-.62 1.4-1.39v-.61h1.16l2.8 7h-21.92zm19.16 12.61c0 .21-.18.39-.39.39h-21.22a.39.39 0 0 1 -.39-.39v-4.61h22z'
            } else if(bedoption==='1 single bed'){
                bedicon = 'm20.99 15.39-1.99-8.45v-5.44c0-.83-.68-1.5-1.5-1.5h-10a1.5 1.5 0 0 0 -1.5 1.5v5.44l-1.99 8.44-.01.12v5.01c0 .66.43 1.2 1.02 1.4-.01.03-.02.06-.02.09v1.5a.5.5 0 0 0 1 0v-1.5h13v1.5a.5.5 0 0 0 1 0v-1.5c0-.03-.01-.06-.02-.09a1.49 1.49 0 0 0 1.02-1.4v-5.01l-.01-.12zm-13.99-13.89a.5.5 0 0 1 .5-.5h9.99c.27 0 .5.23.5.5v4.5h-2v-2.51c.01-.82-.66-1.49-1.48-1.49h-4.02c-.82 0-1.49.67-1.49 1.49v2.51h-2zm8 1.99v4.02a.5.5 0 0 1 -.49.49h-4.02a.5.5 0 0 1 -.49-.49v-4.02c0-.27.22-.49.49-.49h4.02c.27 0 .49.22.49.49zm-8.01 3.63.01-.12h2v .51c0 .82.67 1.49 1.49 1.49h4.02c.82 0 1.49-.67 1.49-1.49v-.51h2l .01.12 1.86 7.88h-14.74l1.86-7.89zm13.01 13.39a.5.5 0 0 1 -.5.49h-14c-.28 0-.5-.22-.5-.49v-4.51h15z'
            }
            arrgs.push(
                <div id='brarrg'>
                    <svg viewBox={'0 0 24 24'} role={'presentation'} focusable={'false'}  >
                        <path d={bedicon} fill-rule={"evenodd"} ></path>
                    </svg>
                    <div>
                        <p id='brtext'>bedroom {i}</p>
                        <p id='bedinroom'>{bedoption}</p>
                    </div>
                </div>
            )
        }
        var displaydesc = {display: this.state.showmoredesc ? 'block': 'none'}
        var displayshowmore = {display: this.state.showmoredesc ? 'none': 'flex'}
        // var displaynextarr = {display:this.state.showscrolltonext? 'block':'none'}
        var arrow = 'm16.29 4.3a1 1 0 1 1 1.41 1.42l-8 8a1 1 0 0 1 -1.41 0l-8-8a1 1 0 1 1 1.41-1.42l7.29 7.29z'
        var movestyle = {transform: `translateX(${this.state.move}%)`}
        
        return(
            <div id='main'>
                <div id='summary'>
                    <div id='tilo'>
                        <div id='title'>{title}</div>
                        <div id='location'>{location}</div>
                    </div>
                    <div id='host'> 
                        { host &&
                            <div >
                                <img src={host.pic}></img>
                                <div id='hostname'>{host.name}</div>   
                            </div>
                        } 
                    </div>                  
                </div>
                <div id='hls'>
                    { highlights &&
                        <div>
                            {Object.keys(highlights).map(hl=>
                                <div id='hl'>
                                    <div id='hlicon'>
                                        {(hl.includes('house')||hl.includes('room')||hl.includes('apartment')) && 
                                            <img src={"./house.png"}></img>
                                        }    
                                        {hl.includes('Self') && 
                                            <img src={"./easy-chkin.png"}></img>
                                        }   
                                        {hl.includes('experience') && 
                                            <img src={"./gt-chkin.png"}></img>
                                        }       
                                        {hl.includes('Superhost') && 
                                            <img src={"./superhost.png"}></img>
                                        } 
                                        {hl.includes('clean') && 
                                            <img src={"./clean.png"}></img>
                                        } 
                                        {hl.includes('location') && 
                                            <img src={"./location.png"}></img>
                                        }                    
                                    </div>
                                    <div id='hltitle'>
                                        {hl}
                                        <div id='hlcontent'>{highlights[hl]}</div>
                                    </div>
                                </div>
                            )}
                        </div>
                    }                   
                </div>
                <div id='linemg'>
                        <div id='line'></div>
                </div>
                <div id='descs'>                    
                { desc &&
                    <div>
                        <div id='generaldesc'>
                             <p id='desccontent'>{desc['General']}</p>
                        </div>
                        <div id='readmore' onClick={e=>this.readmoredesc()} style={displayshowmore}>
                            <div >Read more about this space</div>
                            <svg viewBox={'0 0 18 18'} role={'presentation'} focusable={'false'}  >
                                <path d={arrow} fill-rule={"evenodd"} ></path>
                            </svg>
                        </div>
                        <div id='readmorearea' style={displaydesc}>
                            <div>
                                {Object.keys(desc).slice(1).map(de=>
                                <div id='desctitle'>
                                    {de}
                                        <p id='desccontent'>{desc[de]}</p>
                                    </div>
                                    )}
                            </div>
                            <div id='fakedesclink' >
                                <div >Learn about this number</div>
                            </div>
                            <div id='readless' onClick={e=>this.readlessdesc()} >
                                <div >Hide</div>
                                <svg viewBox={'0 0 18 18'} role={'presentation'} focusable={'false'} >
                                    <path d={arrow} fill-rule={"evenodd"} ></path>
                                </svg>
                            </div>                           
                        </div>   
                        <div id='fakedesclink' >
                            <div >Contact host</div>
                        </div>
                    </div>
                }
                </div>
                <div id='linemg'>
                    <div id='line'></div>
                </div>
                <div id='roomarrg'>
                    <h2>Sleeping arrangements</h2>
                    <div id='arrgline'>
                        <div id='btn-pre' onClick={e=>this.getpre()}>
                            <span>
                                <button type={"button"}>
                                    <svg viewBox={'0 0 18 18'} role={'presentation'} focusable={'false'}  >
                                        <path d={arrow} fill-rule={"evenodd"} ></path>
                                    </svg>
                                </button>
                            </span>
                        </div>
                        <div id='arrg'>
                            <div id='arrgarea' style={movestyle}>
                                {arrgs}
                            </div>
                        </div>
                        <div id='btn-next' onClick={e=>this.getnext()}>
                            <span>
                                <button type={"button"}>
                                    <svg viewBox={'0 0 18 18'} role={'presentation'} focusable={'false'}  >
                                        <path d={arrow} fill-rule={"evenodd"} ></path>
                                    </svg>
                                </button>
                            </span>
                        </div>
                    </div>
                </div>
               
            </div>

        )
    }
}


export default Description;