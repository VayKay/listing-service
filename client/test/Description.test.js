import React from 'react';
import Description from '../component/Description';
import { shallow, configure ,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';


configure({ adapter: new Adapter() });
require('babel-core/register');
require('babel-polyfill');

describe('Description basic rendering test suite', () => {
    
    test('render Description', () => {
        const description = shallow(<Description />) 
        // console.log(description.debug()) 
    });

    test('snapshot', async() => {     
        const description = shallow(<Description />)
        description.setProps({
            title: "Quis ab sint veritatis",
            location: "Schneidermouth",
            host: {
                "name":"Hardy",
                "pic":"http://lorempixel.com/640/480"
            },
            detail: {
                "type":"Hotel room",
                "bedrmnum":1,
                "bathrmnum":1,
                "guestmax":3,
                "bednum":[ "1 single bed" ]    
            },
            highlights: {
                "Private room in hostel":"3 guests\t1 room\t1 bed\t1 bath",
                "Self check-in":"Rerum modi nihil est et.",
                "Hardy is a Superhost":"Iure illo possimus impedit blanditiis ullam laborum omnis.",
                "Sparkling clean":"In necessitatibus id."
            },
            desc: {
                "General":"General description.",
                "Interation with guests":"Description paragraph 2",
                "License or registartion number":"STR-3921741"
            }
        })       
        expect(toJson(description)).toMatchSnapshot();
    });

    test('render listing house titel without crashing', () => {
        const description = shallow(<Description />) 
        expect(description.find('#title').exists()).toBeTruthy() 
    });
    test('render listing house titel with correct text', () => {
        const description = shallow(<Description />) 
        description.setProps({
            title: "Quis ab sint veritatis"
        });
        expect(description.find('#title').text()).toEqual("Quis ab sint veritatis")
    });

    test('render listing house location without crashing', () => {
        const description = shallow(<Description />) 
        expect(description.find('#location').exists()).toBeTruthy() 
    });

    test('render listing house location with correct text', () => {
        const description = shallow(<Description />) 
        description.setProps({
            location: "Schneidermouth"
        });
        expect(description.find('#location').text()).toEqual("Schneidermouth")
    });

    test('host div has 2 children ', () => {
        const description =  mount(<Description />) 
        description.setProps({
            host: {
                "name":"Hardy",
                "pic":"http://lorempixel.com/640/480"
            }
        })       
        // console.log('===',description.debug());
        expect(description.find('#host').children().length).toBe(2)
      });

      test(' highlight div has 4 children ', () => {
        const description =  mount(<Description />) 
        description.setProps({          
            highlights: {
                "Private room in hostel":"3 guests\t1 room\t1 bed\t1 bath",
                "Self check-in":"Rerum modi nihil est et.",
                "Hardy is a Superhost":"Iure illo possimus impedit blanditiis ullam laborum omnis.",
                "Sparkling clean":"In necessitatibus id."
            },         
        })       
        expect(description.find('#hls').children().children().length).toBe(4)
      });
      test('render correct amount of bedrooms in sleeping arrangement area', () => {
        const description =  mount(<Description />) 
        var mockData = {
            detail: {"bedrmnum":2}
        }
        var roomCount = mockData.detail.bedrmnum;
        console.log('count',roomCount)
        description.setProps({
            detail: {
              "bedrmnum":2,
              "bednum" :[ "1 queen bed", "1 king bed"]
            },
        })  
        console.log(description.debug())
        expect(description.find('#arrgarea').children().length).toBe(roomCount)
      });
});
describe('Description animation test suite', () => {

      test('show general description area before click button', () => {
        const description =  mount(<Description />) 
        description.setProps({           
            desc: {
                "General":"General description.",
                "Interation with guests":"Description paragraph 2",
                "License or registartion number":"STR-3921741"
            }          
        })      
        expect(description.state("showmoredesc")).toEqual(false)
      });

      test('show more description area after click button', () => {
        const description =  mount(<Description />) 
        description.setProps({          
            desc: {
                "General":"General description.",
                "Interation with guests":"Description paragraph 2",
                "License or registartion number":"STR-3921741"
            }         
        })
        description.find('#readmore').simulate('click')
        expect(description.state("showmoredesc")).toEqual(true)
      });

      test('doesn\'t show amenity list before click button', () => {
        const description =  mount(<Description />) 
        
        expect(description.state("showstatus")).toEqual(false)
      });

      test('show amenity list after click button', () => {
        const description =  mount(<Description />) 
        description.find('#btn-moreamen').simulate('click')
        expect(description.state("showstatus")).toEqual(true)
      });
      test('show amenity list after click button', () => {
        const description =  mount(<Description />) 
        description.setState({
            showstatus: true
        })   
        description.find('#btn-close').simulate('click')
        expect(description.state("showstatus")).toEqual(false)
      });

      //////
      test('no slide animition before click button in sleeping arrangement area', () => {
        const description =  mount(<Description />) 
        const degree = description.find('#arrgarea').props('style').style.transform.split(/[()%]+/)[1]
        expect(degree).toEqual('0')
      });

      test('slide left after click previous button in sleeping arrangement area', () => {
        const description =  mount(<Description />) 
        const degree = description.find('#arrgarea').props('style').style.transform.split(/[()%]+/)[1]
        description.find('#btn-pre').simulate('click')
        const newDegree = (parseFloat(degree)+(100/3)).toString()+'%'
        expect(description.find('#arrgarea').props('style').style.transform.split(/[()]+/)[1]).toEqual(newDegree)
      });

      test('slide right after click next button in sleeping arrangement area', () => {
        const description =  mount(<Description />) 
        const degree = description.find('#arrgarea').props('style').style.transform.split(/[()%]+/)[1]
        description.find('#btn-next').simulate('click')
        const newDegree = (parseFloat(degree)-(100/3)).toString()+'%'
        expect(description.find('#arrgarea').props('style').style.transform.split(/[()]+/)[1]).toEqual(newDegree)
      });
});