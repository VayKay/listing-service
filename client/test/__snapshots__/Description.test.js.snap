// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description basic rendering test suite snapshot 1`] = `
<div>
  <div
    id="summary"
  >
    <div
      id="tilo"
    >
      <div
        id="title"
      >
        Quis ab sint veritatis
      </div>
      <div
        id="location"
      >
        Schneidermouth
      </div>
    </div>
    <div>
      <div
        id="host"
      >
        <img
          src="http://lorempixel.com/640/480"
        />
        <div
          id="hostname"
        >
          Hardy
        </div>
      </div>
    </div>
  </div>
  <div
    id="hls"
  >
    <div>
      <div
        id="hl"
        key="Private room in hostel"
      >
        <div
          id="hlicon"
        >
          <img
            src="./house.png"
          />
        </div>
        <div
          id="hltitle"
        >
          Private room in hostel
          <div
            id="hlcontent"
          >
            3 guests	1 room	1 bed	1 bath
          </div>
        </div>
      </div>
      <div
        id="hl"
        key="Self check-in"
      >
        <div
          id="hlicon"
        >
          <img
            src="./easy-chkin.png"
          />
        </div>
        <div
          id="hltitle"
        >
          Self check-in
          <div
            id="hlcontent"
          >
            Rerum modi nihil est et.
          </div>
        </div>
      </div>
      <div
        id="hl"
        key="Hardy is a Superhost"
      >
        <div
          id="hlicon"
        >
          <img
            src="./superhost.png"
          />
        </div>
        <div
          id="hltitle"
        >
          Hardy is a Superhost
          <div
            id="hlcontent"
          >
            Iure illo possimus impedit blanditiis ullam laborum omnis.
          </div>
        </div>
      </div>
      <div
        id="hl"
        key="Sparkling clean"
      >
        <div
          id="hlicon"
        >
          <img
            src="./clean.png"
          />
        </div>
        <div
          id="hltitle"
        >
          Sparkling clean
          <div
            id="hlcontent"
          >
            In necessitatibus id.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    id="linemg"
  >
    <div
      id="line"
    />
  </div>
  <div
    id="descs"
  >
    <div>
      <div
        id="generaldesc"
      >
        <p
          id="desccontent"
        >
          General description.
        </p>
      </div>
      <div
        id="readmore"
        onClick={[Function]}
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div>
          Read more about this space
        </div>
        <svg
          focusable="false"
          role="presentation"
          viewBox="0 0 18 18"
        >
          <path
            d="m16.29 4.3a1 1 0 1 1 1.41 1.42l-8 8a1 1 0 0 1 -1.41 0l-8-8a1 1 0 1 1 1.41-1.42l7.29 7.29z"
          />
        </svg>
      </div>
      <div
        id="readmorearea"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div>
          <div
            id="desctitle"
            key="Interation with guests"
          >
            Interation with guests
            <p
              id="desccontent"
            >
              Description paragraph 2
            </p>
          </div>
          <div
            id="desctitle"
            key="License or registartion number"
          >
            License or registartion number
            <p
              id="desccontent"
            >
              STR-3921741
            </p>
          </div>
        </div>
        <div
          id="fakedesclink"
        >
          <div>
            Learn about this number
          </div>
        </div>
        <div
          id="readless"
          onClick={[Function]}
        >
          <div>
            Hide
          </div>
          <svg
            focusable="false"
            role="presentation"
            viewBox="0 0 18 18"
          >
            <path
              d="m16.29 4.3a1 1 0 1 1 1.41 1.42l-8 8a1 1 0 0 1 -1.41 0l-8-8a1 1 0 1 1 1.41-1.42l7.29 7.29z"
            />
          </svg>
        </div>
      </div>
      <div
        id="fakedesclink"
      >
        <div>
          Contact host
        </div>
      </div>
    </div>
  </div>
  <div
    id="linemg"
  >
    <div
      id="line"
    />
  </div>
  <Amenity
    showamenitylist={[Function]}
  />
  <div
    id="linemg"
  >
    <div
      id="line"
    />
  </div>
  <div
    id="roomarrg"
  >
    <h2>
      Sleeping arrangements
    </h2>
    <div
      id="arrgline"
    >
      <div
        id="btn-pre"
        onClick={[Function]}
      >
        <span>
          <button
            type="button"
          >
            <svg
              focusable="false"
              role="presentation"
              viewBox="0 0 18 18"
            >
              <path
                d="m16.29 4.3a1 1 0 1 1 1.41 1.42l-8 8a1 1 0 0 1 -1.41 0l-8-8a1 1 0 1 1 1.41-1.42l7.29 7.29z"
              />
            </svg>
          </button>
        </span>
      </div>
      <div
        id="arrg"
      >
        <div
          id="arrgarea"
          style={
            Object {
              "transform": "translateX(0%)",
            }
          }
        >
          <div
            id="brarrg"
            key="1"
          >
            <svg
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="m20.99 15.39-1.99-8.45v-5.44c0-.83-.68-1.5-1.5-1.5h-10a1.5 1.5 0 0 0 -1.5 1.5v5.44l-1.99 8.44-.01.12v5.01c0 .66.43 1.2 1.02 1.4-.01.03-.02.06-.02.09v1.5a.5.5 0 0 0 1 0v-1.5h13v1.5a.5.5 0 0 0 1 0v-1.5c0-.03-.01-.06-.02-.09a1.49 1.49 0 0 0 1.02-1.4v-5.01l-.01-.12zm-13.99-13.89a.5.5 0 0 1 .5-.5h9.99c.27 0 .5.23.5.5v4.5h-2v-2.51c.01-.82-.66-1.49-1.48-1.49h-4.02c-.82 0-1.49.67-1.49 1.49v2.51h-2zm8 1.99v4.02a.5.5 0 0 1 -.49.49h-4.02a.5.5 0 0 1 -.49-.49v-4.02c0-.27.22-.49.49-.49h4.02c.27 0 .49.22.49.49zm-8.01 3.63.01-.12h2v .51c0 .82.67 1.49 1.49 1.49h4.02c.82 0 1.49-.67 1.49-1.49v-.51h2l .01.12 1.86 7.88h-14.74l1.86-7.89zm13.01 13.39a.5.5 0 0 1 -.5.49h-14c-.28 0-.5-.22-.5-.49v-4.51h15z"
              />
            </svg>
            <div>
              <p
                id="brtext"
              >
                bedroom 
                1
              </p>
              <p
                id="bedinroom"
              >
                1 single bed
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        id="btn-next"
        onClick={[Function]}
      >
        <span>
          <button
            type="button"
          >
            <svg
              focusable="false"
              role="presentation"
              viewBox="0 0 18 18"
            >
              <path
                d="m16.29 4.3a1 1 0 1 1 1.41 1.42l-8 8a1 1 0 0 1 -1.41 0l-8-8a1 1 0 1 1 1.41-1.42l7.29 7.29z"
              />
            </svg>
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    id="poplistback"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      id="poplist"
    >
      <div
        id="poplisthead"
      >
        <button
          id="btn-close"
          onClick={[Function]}
        >
          <svg
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="m23.25 24c-.19 0-.38-.07-.53-.22l-10.72-10.72-10.72 10.72c-.29.29-.77.29-1.06 0s-.29-.77 0-1.06l10.72-10.72-10.72-10.72c-.29-.29-.29-.77 0-1.06s.77-.29 1.06 0l10.72 10.72 10.72-10.72c.29-.29.77-.29 1.06 0s .29.77 0 1.06l-10.72 10.72 10.72 10.72c.29.29.29.77 0 1.06-.15.15-.34.22-.53.22"
            />
          </svg>
        </button>
      </div>
      <div
        id="poplisttitle"
      >
        Amenities
      </div>
      <div
        id="poplistcontent"
      />
    </div>
  </div>
</div>
`;
